lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -2043,14 +2062,19 @@ class BuildControl : public calyx::FuncOpPartialLoweringPattern {
       } else if (auto *parSchedPtr = std::get_if<ParScheduleable>(&group)) {
         auto parOp = parSchedPtr->parOp;
         auto calyxParOp = rewriter.create<calyx::ParOp>(parOp.getLoc());
-        for (auto &innerBlock : parOp.getRegion().getBlocks()) {
-          rewriter.setInsertionPointToEnd(calyxParOp.getBodyBlock());
-          auto seqOp = rewriter.create<calyx::SeqOp>(parOp.getLoc());
-          rewriter.setInsertionPointToEnd(seqOp.getBodyBlock());
-          if (LogicalResult res = scheduleBasicBlock(
-                  rewriter, path, seqOp.getBodyBlock(), &innerBlock);
-              res.failed())
-            return res;
+        for (auto &op : parOp.getBody()->getOperations()) {
+          if (auto execRegion = dyn_cast<scf::ExecuteRegionOp>(&op)) {
```

Comment:
Nit: There's 4 levels of nesting here. Can we do some early 'break' statements to reduce this?